src/cms-components/core/page.js

```diff
@@ -118,10 +118,12 @@ export default class CmsPage extends React.Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    if (this.props.request.path !== prevProps.request.path) {
+    if (this.props.request.path !== prevProps.request.path && !this.props.pageModel) {
```

Comment:
This may break navigation within an isomorphic application. When a user has already loaded a page and navigates to another, we may replace a request.

Or did you mean something like?
```
this.props.request.path !== prevProps.request.path || !this.props.pageModel
```

Reply:
Thanks for your response, I will try to explain my toughts on this.

In case of our Isomorphic application (Nextjs), to make use of this component we first need to pass the pageModel on initial load, which will run the `updatePageModel` function in `componentDidMount` and therefore the data will be present during server side rendering.

After that, when the user moves to a new page (and changes the request) we need to (in the current situation) stop NextJs from fetching the pageModel and give that logic to this component. This will prevent us from using next/router event changes, as they are only applicable to everything that happens in getInitialProps and this function will run server-side or client side when the url changes. Ideally we would like to make use of this feature of NextJS.

The SSR example will actually run two fetches, one in NextJS and one in this component.

A regular client side rendered version will resolve in `true` in above statement, as the request has changed and the pageModel is never passed as a property.

Am I missing a use case?

Reply:
The problem with this condition is that when we update the `request` property of `CmsPage`, and the previous value of `pageModel` property was `undefined`, the component will never get updated.

This can happen when we initialize `CmsPage` without properties. In this case, React Router or Next Routes will render a new instance of the component and React will just update these props after diffing.

The correct way will be to update the page model state whenever the `pageModel` property is getting updated. And in case if it's not, we can fetch a new model on the path change.

Reply:
Thank you so much @dokmic , 
Nice solution!